src/Argu/Parsers/Cli.fs

```diff
@@ -236,8 +241,7 @@ let rec private parseCommandLinePartial (state : CliParseState) (argInfo : Union
             | ListParam(existential, field) ->
                 ignore <| existential.Accept { new IFunc<bool> with
                     member __.Invoke<'T>() =
-                        let args = new ResizeArray<'T> ()
-                        let rec gather isFirst =
+                        let rec gather isFirst args =
```

Comment:
I don't think using a list accumulator here improves clarity much. The code here is mostly impure, so we can save a few unnecessary allocations by using an impure aggregator.

Reply:
I certainly don't mind either way, but important that what is accumulated is of type `string`, as opposed to `'T`.

Reply:
Oh, I missed that. Why is this significant?

Reply:
Actually, it's not that important now I think about it some more. Nevertheless, the new logic needs to determine whether to assign the token as an Unrecognized (type string), or Main (type 'T). It could be cleaner to collect them as strings and then parse to 'T (again) if needs be. Otherwise, we'd need to convert from 'T back to string again if the arguments are adjudged to be Unrecognized. I assume the latter would work fine, but didn't try it.

Reply:
Makes sense. As long as no regressions happen should be fine.

Reply:
It looks like all tests are passing. 

Can I expect these changes in a v5.6 release at some point?

Reply:
> It looks like all tests are passing.

Tests are hardly comprehensive, and interop of all the different features is hardly even spec'ed (as you found out). I will need to trust you (as the person to have most recently seen this code) that it will mostly keep behaving as expected :-)

Reply:
I'm happy to be trusted on this. I will of course revisit if any of my changes cause problems to any examples out there. But perhaps we should try and add some more tests. Next time I contribute something to this repo, I'll expand the tests.